src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -248,6 +251,44 @@ internal static class InheritanceMarginServiceHelper
                     .Concat(overridingMemberItems));
         }
 
+        private static ImmutableArray<ISymbol> GetImplementingSymbolsForTypeMember(
+            ISymbol memberSymbol,
+            ImmutableArray<ISymbol> overridingSymbols)
+        {
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol)
+            {
+                // Use a hashset builder because there could be duplicates.
+                // For example:
+                // interface IBar { void Foo(); }
+                // class Bar : IBar { public virtual void Foo() { } }
+                // class Bar2 : Bar, IBar { public override void Foo() { } }
+                // Here IBar.Foo() would be added twice as an implementing members.
+                // The order is not important, because UI would sort it based on display name.
+                using var _ = PooledHashSet<ISymbol>.GetInstance(out var builder);
+
+                // 1. Add the direct implementing symbols in interfaces.
+                var directImplementingSymbols = memberSymbol.ExplicitOrImplicitInterfaceImplementations()
+                    .SelectAsArray(symbol => symbol.OriginalDefinition);
+                builder.AddRange(directImplementingSymbols);
```

Comment:
do you need to Distinct tehse?

Reply:
The builder is a set, so all the item should be distinct.